server/routes/contacts/create/enter-name/createContactEnterNameController.ts

```diff
@@ -6,10 +6,37 @@ import { CreateContactEnterNameSchemaType } from './createContactEnterNameSchema
 export default class EnterNameController implements PageHandler {
   public PAGE_NAME = Page.CREATE_CONTACT_NAME_PAGE
 
+  private TITLES = [
+    { value: '', text: '' },
+    { value: 'DAME', text: 'Dame' },
+    { value: 'FR', text: 'Father' },
+    { value: 'LORD', text: 'Lord' },
+    { value: 'MS', text: 'Ms' },
+    { value: 'RABBI', text: 'Rabbi' },
+    { value: 'REV', text: 'Reverend' },
+    { value: 'SIR', text: 'Sir' },
+    { value: 'SR', text: 'Sister' },
+    { value: 'MR', text: 'Mr' },
+    { value: 'BR', text: 'Brother' },
+    { value: 'DR', text: 'Dr' },
+    { value: 'LADY', text: 'Lady' },
+    { value: 'MISS', text: 'Miss' },
+    { value: 'MRS', text: 'Mrs' },
+    { value: 'IMAM', text: 'Imam' },
+  ]
+
   GET = async (req: Request, res: Response): Promise<void> => {
     const { journeyId } = req.params
     const journey = req.session.createContactJourneys[journeyId]
-    res.render('pages/contacts/create/enterName', { journey })
+
+    const viewModel = {
+      journey,
+      titleOptions: this.getSelectedTitleOptions(res.locals?.formResponses?.title ?? journey?.names?.title),
+      lastName: res.locals?.formResponses?.lastName ?? journey?.names?.lastName,
+      firstName: res.locals?.formResponses?.firstName ?? journey?.names?.firstName,
+      middleName: res.locals?.formResponses?.middleName ?? journey?.names?.middleName,
```

Comment:
Wondering about this part? Nunjucks templates can directly access both the journey and formResponses, so can do this logic on setting values   e.g.   `value = res.locals?.formResponses?.lastName ?? journey?.names?.lastName`, though it is fairly neat to just provide a `viewModel` for each form.  Trade off between a) Building a view model in handlers with extra code, or b) Reducing code in handlers, but increasing it in nunjucks. Ultimately, a choice..

Reply:
Thanks, I was also unsure whether to do it here or in the template itself. I had 1 eye on DOB where we have to convert back from a date into the day, month, year fields and that felt like it'd be much easier in the controller.
Also search is currently using this pattern.